pkg/apis/core/v1alpha1/types_overridepolicy.go

```diff
@@ -81,12 +81,46 @@ type TargetClusters struct {
 	ClusterAffinity []ClusterSelectorTerm `json:"clusterAffinity,omitempty"`
 }
 
+// Overriders contains a list of override patch
+// The order in which the override patches take effect is:
+// - Image
+// - JsonPatch
 type Overriders struct {
+	Image []ImageOverrider `json:"image,omitempty"`
 	// JsonPatch specifies overriders in a syntax similar to RFC6902 JSON Patch.
 	// +optional
 	JsonPatch []JsonPatchOverrider `json:"jsonpatch,omitempty"`
 }
 
+type ImageOverrider struct {
+	// ContainerName is ignored when ImagePath is set
+	// If null, the image override rule applies to all containers.
+	// If specified, it should match one container or initContainer name in pod template.
+	// +optional
+	ContainerName string `json:"containerName,omitempty"`
+
+	// ImagePath indicates the path of target image field.
+	// Defaults to empty, the system will automatically detect image fields if the resource type is
+	// Pod, CronJob, Deployment, StatefulSet, DaemonSet or Job
+	// If not empty, the image override rule applies to specify image
+	// +optional
+	ImagePath string `json:"imagePath,omitempty"`
+
+	// ImageComponent is part of image name.
+	// +kubebuilder:validation:Enum=registry;repository;tag;digest
+	// +required
```

Comment:
There's no such marker.

Reply:
See the full list of kubebuilder markers https://book.kubebuilder.io/reference/markers/crd-validation

Reply:
Additionally, the `OverridePolicy` struct already has the `+kubebuilder:valudation:Required` marker, and it seems like the marker is passed down recursively to the struct's fields. Thus all fields would be required by default and we only need to explicitly mark the optional fields.

Reply:
The code has been updated.